### PR TITLE
drivers: watchdog: Driver for Infineon watchdog

### PR DESCRIPTION
**Pull request by sreeramIfx**
Initial version of the driver for Infineon CAT1 devices

### CODE REVIEW

File: drivers/watchdog/wdt_ifx_cat1.c [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+	return 0;
+}
+
+static int ifx_cat1_wdt_disable(const struct device *dev)
+{
+	struct ifx_cat1_wdt_data *dev_data = dev->data;
+
+	Cy_WDT_MaskInterrupt();
+	irq_disable(DT_INST_IRQN(0));
```
Comment (npal-cy) [inline drivers/watchdog/wdt_ifx_cat1.c:69]: #ifdef IFX_CAT1_WDT_IS_IRQ_EN
        Cy_WDT_MaskInterrupt();
	irq_disable(DT_INST_IRQN(0));
#endif

Comment (sreeramIfx) [reply drivers/watchdog/wdt_ifx_cat1.c:70]: updated..

File: drivers/watchdog/Kconfig.ifx_cat1 [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Infineon CAT1 Watchdog configuration options
+
+# Copyright (c) 2023 Cypress Semiconductor Corporation (an Infineon company) or
+# an affiliate of Cypress Semiconductor Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config WATCHDOG_INFINEON_CAT1
+	bool "Infineon CAT1 Watchdog Driver"
```
Comment (katsuster) [inline drivers/watchdog/Kconfig.ifx_cat1:9]: Almost others starts with "WDT_" so I think it's better to change WDT_INFINEON_CAT1.

File: CODEOWNERS [hunk 3]
```diff
@@ -474,6 +474,7 @@
 /drivers/watchdog/*rpi_pico*              @thedjnK
 /drivers/watchdog/*dw*                    @softwarecki
 /drivers/wifi/                            @rlubos @tbursztyka
+/drivers/watchdog/*ifx*                   @sreeramIfx
 /drivers/wifi/esp_at/                     @mniestroj
```
Comment (katsuster) [inline CODEOWNERS:478]: Why here? 1 line up is better.

File: drivers/watchdog/Kconfig.ifx_cat1 (hunk 2 above)
Comment (sreeramIfx) [reply drivers/watchdog/Kconfig.ifx_cat1:9]: updated..

File: CODEOWNERS (hunk 3 above)
Comment (sreeramIfx) [reply CODEOWNERS:478]: updated..

**Review by katsuster — DISMISSED** [review]

**Review by npal-cy — APPROVED** [review]

**Review by katsuster — APPROVED** [review]

**Review by ifyall — APPROVED** [review]

**Pull request merged**